gobblin-data-management/src/main/java/org/apache/gobblin/data/management/retention/DatasetCleaner.java

```diff
@@ -164,9 +171,11 @@ public void onFailure(Throwable throwable) {
         }
 
         @Override
-        public void onSuccess(Void arg0) {
+        public void onSuccess(Integer datasetsDeleted) {
           DatasetCleaner.this.finishCleanSignal.get().countDown();
-          LOG.info("Successfully cleaned: " + dataset.datasetURN());
+          DatasetRetentionSummary summary = new DatasetRetentionSummary(dataset.datasetURN(), datasetsDeleted, true);
```

Comment:
is it certain we'd only ever overwrite the summary, rather than potentially needing to add this new count to whatever accumulating sum was already tracked for the same dataset URN?